dashboard/pages/daily_report.py

```diff
@@ -0,0 +1,95 @@
+"""
+creates a page in the dashboard that allows users to subscribe to a summary report that
+is emailed to them every day. These reports are also stored in an S3 bucket
+"""
+import re
+from os import environ
+from dotenv import load_dotenv
+import streamlit as st
+from psycopg2 import connect
+from psycopg2.extras import RealDictCursor
+from psycopg2.extensions import connection, cursor
+
+def get_db_connection() -> connection:
+    """return a database connection"""
+    try:
+        return connect(
+            host=environ['DB_HOST'],
+            dbname=environ['DB_NAME'],
+            user=environ['DB_USERNAME'],
+            password=environ['DB_PASSWORD'],
+            port=environ['DB_PORT']
+        )
+    except Exception as e: # pylint: disable=broad-exception-caught
+        print(e)
+        return None
+
+
+def get_db_cursor(conn: connection) -> cursor:
+    """return a cursor object based on a given connection"""
+    try:
+        return conn.cursor(cursor_factory=RealDictCursor)
+    except Exception as e: # pylint: disable=broad-exception-caught
+        st.write(e)
+        return None
+
+
+def is_email_already_subscribed(email: str, conn: connection) -> bool:
+    """check if the email given has already subscribed to the database"""
+    with get_db_cursor(conn) as curs:
+        curs.execute("SELECT COUNT(*) FROM subscriber WHERE email=%s", (email, ))
+        res = curs.fetchone()
+    return res["count"] == 1
+
+
+def is_email_valid(email: str) -> bool:
+    """return a bool based on whether this is a valid email"""
+    email_pattern = re.compile(r"^[\w\-\.]+@([\w\-]+\.)+[\w\-]{2,4}$")
```

Comment:
```suggestion
    email_pattern = re.compile(r"[\w\-\.]+@([\w\-]+\.)+[\w\-]{2,4}$")
```

Reply:
https://docs.python.org/3/library/re.html#search-vs-match
since match is used, using ^ at the beginning is redundant unless you want to use search

Reply:
You can also use fullmatch and remove ^ and $